Build error when compiling from source with -DDISABLE_EXTENSION_LOAD=1

### What happens?

Hello,

there is a build error when compiling DuckDB from source with the option -DDISABLE_EXTENSION_LOAD=1:

```
[ 52%] Building CXX object src/main/extension/CMakeFiles/duckdb_main_extension.dir/ub_duckdb_main_extension.cpp.o
cd /Users/ddeleo/workspace/duckdb/build/src/main/extension && /Library/Developer/CommandLineTools/usr/bin/c++ -DDUCKDB -DDUCKDB_BUILD_LIBRARY -DDUCKDB_DISABLE_EXTENSION_LOAD -DDUCKDB_EXTENSION_AZURE_LINKED=1 -DDUCKDB_EXTENSION_HTTPFS_LINKED=1 -DDUCKDB_EXTENSION_ICEBERG_LINKED=1 -DDUCKDB_EXTENSION_PARQUET_LINKED=1 -DDUCKDB_MAIN_LIBRARY -DGENERATED_EXTENSION_HEADERS=1 -I/Users/ddeleo/workspace/duckdb/src/include -I/Users/ddeleo/workspace/duckdb/third_party/fsst -I/Users/ddeleo/workspace/duckdb/third_party/fmt/include -I/Users/ddeleo/workspace/duckdb/third_party/hyperloglog -I/Users/ddeleo/workspace/duckdb/third_party/fastpforlib -I/Users/ddeleo/workspace/duckdb/third_party/skiplist -I/Users/ddeleo/workspace/duckdb/third_party/fast_float -I/Users/ddeleo/workspace/duckdb/third_party/re2 -I/Users/ddeleo/workspace/duckdb/third_party/miniz -I/Users/ddeleo/workspace/duckdb/third_party/utf8proc/include -I/Users/ddeleo/workspace/duckdb/third_party/concurrentqueue -I/Users/ddeleo/workspace/duckdb/third_party/pcg -I/Users/ddeleo/workspace/duckdb/third_party/tdigest -I/Users/ddeleo/workspace/duckdb/third_party/mbedtls/include -I/Users/ddeleo/workspace/duckdb/third_party/jaro_winkler -I/Users/ddeleo/workspace/duckdb/third_party/yyjson/include -I/Users/ddeleo/workspace/duckdb/extension -I/Users/ddeleo/workspace/duckdb/extension/httpfs/include -I/Users/ddeleo/workspace/duckdb/build/_deps/azure_extension_fc-src/src/include -I/Users/ddeleo/workspace/duckdb/build/_deps/iceberg_extension_fc-src/src/include -I/Users/ddeleo/workspace/duckdb/extension/parquet/include -I/Users/ddeleo/workspace/duckdb/src/main/extension/../../../third_party/httplib -I/Users/ddeleo/workspace/duckdb/build/codegen/include -O3 -DNDEBUG -O3 -DNDEBUG   -std=c++11 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk -fPIC -MD -MT src/main/extension/CMakeFiles/duckdb_main_extension.dir/ub_duckdb_main_extension.cpp.o -MF CMakeFiles/duckdb_main_extension.dir/ub_duckdb_main_extension.cpp.o.d -o CMakeFiles/duckdb_main_extension.dir/ub_duckdb_main_extension.cpp.o -c /Users/ddeleo/workspace/duckdb/build/src/main/extension/ub_duckdb_main_extension.cpp
In file included from /Users/ddeleo/workspace/duckdb/build/src/main/extension/ub_duckdb_main_extension.cpp:4:
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_install.cpp:351:2: error: use of undeclared identifier 'duckdb_httplib'
        duckdb_httplib::Client cli(url_base.c_str());
        ^
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_install.cpp:353:3: error: use of undeclared identifier 'cli'
                cli.set_logger(http_logger->GetLogger<duckdb_httplib::Request, duckdb_httplib::Response>());
                ^
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_install.cpp:353:41: error: use of undeclared identifier 'duckdb_httplib'
                cli.set_logger(http_logger->GetLogger<duckdb_httplib::Request, duckdb_httplib::Response>());
                                                      ^
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_install.cpp:356:2: error: use of undeclared identifier 'duckdb_httplib'
        duckdb_httplib::Headers headers = {
        ^
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_install.cpp:359:13: error: use of undeclared identifier 'cli'
        auto res = cli.Get(url_local_part.c_str(), headers);
                   ^
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_install.cpp:368:22: error: use of undeclared identifier 'duckdb_httplib'
                if (res.error() == duckdb_httplib::Error::Success) {
                                   ^
In file included from /Users/ddeleo/workspace/duckdb/build/src/main/extension/ub_duckdb_main_extension.cpp:5:
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_load.cpp:126:24: error: use of undeclared identifier 'ComputeSHA256FileSegment'
                threads.emplace_back(ComputeSHA256FileSegment, &handle, splits[i], splits[i + 1], &hash_chunks[i]);
                                     ^
/Users/ddeleo/workspace/duckdb/src/main/extension/extension_load.cpp:146:2: error: use of undeclared identifier 'ComputeSHA256String'
        ComputeSHA256String(hash_concatenation, &two_level_hash);
        ^
8 errors generated.
```

on MacOS M1, clang v.15, tried on DuckDB v.1.0.0 and tip of master at fd883da87821a3e4a60169df7cba49f29d7a252e.



### To Reproduce

Compile the source with the flag `-DDISABLE_EXTENSION_LOAD=1`:
```
cd <duckdb>/build
cmake -DCMAKE_INSTALL_PREFIX=$(pwd)/install \
      -DCMAKE_BUILD_TYPE=Release \
      -DCMAKE_TOOLCHAIN_FILE=${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake \
      -DVCPKG_MANIFEST_DIR=$(pwd)/extension_configuration \
      -DVCPKG_VERBOSE=1 \
      -DDISABLE_EXTENSION_LOAD=1 \
      ..
```

 In my case the httpfs, azure and iceberg extensions are loaded.

### OS:

MacOS Sonoma v14.4.1 

### DuckDB Version:

master

### DuckDB Client:

n/a

### Full Name:

Dean De Leo

### Affiliation:

RelationalAI

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a source build

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

--- a/src/main/extension/extension_install.cpp
+++ b/src/main/extension/extension_install.cpp
@@ -335,6 +335,7 @@ static unique_ptr<ExtensionInstallInfo> DirectInstallExtension(DBConfig &config,
 	return make_uniq<ExtensionInstallInfo>(info);
 }
 
+#ifndef DUCKDB_DISABLE_EXTENSION_LOAD
 static unique_ptr<ExtensionInstallInfo> InstallFromHttpUrl(DBConfig &config, const string &url,
                                                            const string &extension_name, const string &temp_path,
                                                            const string &local_extension_path, bool force_install,
@@ -506,6 +507,7 @@ static void ThrowErrorOnMismatchingExtensionOrigin(FileSystem &fs, const string 
 		}
 	}
 }
+#endif // DUCKDB_DISABLE_EXTENSION_LOAD
 
 unique_ptr<ExtensionInstallInfo>
 ExtensionHelper::InstallExtensionInternal(DBConfig &config, FileSystem &fs, const string &local_path,

--- a/src/main/extension/extension_load.cpp
+++ b/src/main/extension/extension_load.cpp
@@ -32,6 +32,7 @@ static T LoadFunctionFromDLL(void *dll, const string &function_name, const strin
 	}
 	return (T)function;
 }
+#endif
 
 static void ComputeSHA256String(const string &to_hash, string *res) {
 	// Invoke MbedTls function to actually compute sha256
@@ -57,7 +58,6 @@ static void ComputeSHA256FileSegment(FileHandle *handle, const idx_t start, cons
 
 	*res = state.Finalize();
 }
-#endif
 
 static string FilterZeroAtEnd(string s) {
 	while (!s.empty() && s.back() == '\0') {

